src/components/Authentication/Authenticate.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react'
+import { Auth } from "aws-amplify";
+import Login from './Login';
+import Logout from './Logout';
+import { setUser } from '../../actions/UserActions'
+import { connect } from "react-redux";
+
+export class Authenticate extends Component {
+
+    getAuthenticatedUser(){
+        Auth.currentAuthenticatedUser()
+            .then(user => {
+                const email = user.attributes.email
+                if (email.substring(email.indexOf("@") + 1, email.length) === 'ubcbiztech.com') {
+                    this.props.setUser(user)
+                }
+                else {
+                    Auth.signOut()
+                    alert('You must use a ubcbiztech.com email')
+                }
+            })
+            .catch(() => console.log("Not signed in"))
+    }
+
+    componentDidMount() {
+        // Check if there is an authenticated user
+        // After authenticating with Google, Auth redirects to localhost:3000/login/ 
+        // There is a time delay where Auth.currentAuthenticatedUser() returns the wrong value. Hence setTimeout()
+        if (window.location.pathname === '/login/'){
+            setTimeout(() => this.getAuthenticatedUser(), 600)
```

Comment:
I wasn't able to pinpoint the reason why the call returns an initial invalid value. There might a cache may be involved but I couldn't find anything. 0.6s was the shortest I could get it on my machine while still consistently getting correct results, and I have no idea if that will work on other machines or internet connections.

Reply:
Tested on my machine and login seems a little buggy. 

Upon inputting my login information and logging in, it returns me to the login screen and allows me to click `SIGN IN WITH GOOGLE` again and displays a reload; however, nothing happens. After some period of time I get sent to the main page. Able to reproduce consistently

Reply:
If it is a fail-safe to prevent insecure connections from entering the user pool (or something along those lines), could create a listener to `pushState` when notified that a valid response is returned

Reply:
We aren't subscribed to any callback or hook that would notify us of a valid response, but we could use polling. 

After logging in is it redirecting you to localhost:3000/login?

Reply:
After authenticating you should see `SIGN IN WITH GOOGLE` for exactly 600ms and then the page should reload and take you to the main page. Maybe we can redirect to a different page that is not the login page so the `SIGN IN` button isn't visible (and confusing)